[nanojsonc] update 1.2.0

saadshams opened the pull request:
- [x ] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x ] SHA512s are updated for each updated download.
- [x ] The "supports" clause reflects platforms that may be fixed by this new version.
- [x ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x ] Any patches that are no longer applied are deleted from the port's directory.
- [x ] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x ] Only one version is added to each modified port's versions file.

saadshams committed:
update 1.1.0

saadshams committed:
update

saadshams commented:
Unable to upgrade the port to 1.2.0, please support.

saadshams commented:
@LilyWangLL Please guide on what's missing?

LilyWangLL commented on ports/nanojsonc/portfile.cmake:
```diff
@@ -4,7 +4,7 @@ vcpkg_from_github(
         REPO open-source-patterns/nanojsonc
         REF "${VERSION}"
-        SHA512 4cb73a0dc42bc6dbc106ed7bf7d22dbbadf3d92d2055d4b96990b62822978c09e580a87ca1666cf0b915b994fada6a1f6b803eb98d8da6b021a0a2c410d538ff
+        SHA512 8fe36b742b994598d6f5ae30433245f63da5378ff6866f593cf53916e2b2fdc5d85ddc88d20908fd1166c3568f62e2a8dd837270855cb603dad88246a726be51
```
```suggestion
        SHA512 a83f38c67222d227faf352495b174a16ff6d5a0a45e9d6eda499ca6e8978d2dba184f820c7b726d80468a11fee02460cbd8b189d69a46716bd01c46e7b0607fd
```

saadshams replied:
I wonder how did you get this SHA512
What I did, is clicked "Download ZIP" from the code tab of the repository and ran the following command in the terminal

`shasum -a 512 nanojsonc-main.zip
`and I got
`0cfa74eb7fcf4ca43a2dedcf5d13985e74869ae0966c674bad7c8dceb1fb7a0176f55a8520e1c1e0be61c646881315af8adaefba76c59dc4cba7ee1724e88f4e`

LilyWangLL commented:
> Installing 3/3 nanojsonc:x64-windows@1.2.0...
> Building nanojsonc:x64-windows@1.2.0...
> -- Note: nanojsonc only supports static library linkage. Building static library.
> Asset cache miss; downloading from https://github.com/open-source-patterns/nanojsonc/archive/1.2.0.tar.gz
> Downloading open-source-patterns-nanojsonc-1.2.0.tar.gz
> Failed to download open-source-patterns-nanojsonc-1.2.0.tar.gz.
> If you are using a proxy, please ensure your proxy settings are correct.
> Possible causes are:
> 1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable to `https//address:port`.
> This is not correct, because `https://`/ prefix claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr, etc...) is an HTTP proxy.
> Try setting `http://address:port`/ to both HTTP_PROXY and HTTPS_PROXY instead.
> 2. If you are using Windows, vcpkg will automatically use your Windows IE Proxy Settings set by your proxy software. See, https://github.com/microsoft/vcpkg-tool/pull/77
> The value set by your proxy might be wrong, or have same `https://`/ prefix issue.
> 3. Your proxy's remote server is our of service.
> If you've tried directly download the link, and believe this is not a temporay download server failure, please submit an issue at https://github.com/Microsoft/vcpkg/issues
> to report this upstream download server failure.
> error: https://vcpkgassetcachewus.blob.core.windows.net/cache/8fe36b742b994598d6f5ae30433245f63da5378ff6866f593cf53916e2b2fdc5d85ddc88d20908fd1166c3568f62e2a8dd837270855cb603dad88246a726be51?*** SECRET ***: failed: status code 404
> error: File does not have the expected hash:
> url: https://github.com/open-source-patterns/nanojsonc/archive/1.2.0.tar.gz
> File: D:\downloads\open-source-patterns-nanojsonc-1.2.0.tar.gz.8340.part
> Expected hash: 8fe36b742b994598d6f5ae30433245f63da5378ff6866f593cf53916e2b2fdc5d85ddc88d20908fd1166c3568f62e2a8dd837270855cb603dad88246a726be51
> Actual hash: a83f38c67222d227faf352495b174a16ff6d5a0a45e9d6eda499ca6e8978d2dba184f820c7b726d80468a11fee02460cbd8b189d69a46716bd01c46e7b0607fd

LilyWangLL commented:
You could get the failure log from https://dev.azure.com/vcpkg/public/_build/results?buildId=111732&view=logs&jobId=7922e5c4-0103-5f8f-ad17-45ce9bb98e80&j=7922e5c4-0103-5f8f-ad17-45ce9bb98e80&t=8ceef4ed-9be5-594f-e707-4e1dd58a3d21.

LilyWangLL commented:
Please use command `.\vcpkg x-add-version nanojsonc --overwrite-version` to update the version file after you commit the new changes.

saadshams committed:
Update ports/nanojsonc/portfile.cmake

Co-authored-by: Lily Wang <94091114+LilyWangLL@users.noreply.github.com>

saadshams commented:
@LilyWangLL, sorry for the delay on this. I'm on a mac, for some reason I don't have `./vcpkg` command available

`.\vcpkg x-add-version nanojsonc --overwrite-version`

saadshams commented:
@LilyWangLL I have updated `SHA512` and added version programmatically after committing changes.  

`.\vcpkg x-add-version nanojsonc --overwrite-version`

LilyWangLL committed:
update version files

LilyWangLL committed:
update versions/baseline.json

LilyWangLL commented on ports/nanojsonc/vcpkg.json:
```diff
@@ -1,10 +1,11 @@
   "description": "Event-Driven JSON Parser for C",
   "homepage": "https://github.com/open-source-patterns/nanojsonc",
   "license": "BSD-3-Clause",
+  "supports": "windows | linux | osx | ios | android | uwp | xbox | emscripten | wasm32",
```
Which platforms are currently not supported by this port? Stating unsupported platforms shortens the expression.

saadshams replied:
the library is in the plain vanilla c and is supported by all aforementioned platforms.

dg0yt replied:
Easy:
```suggestion
```
= no restrictions

saadshams committed:
Update ports/nanojsonc/vcpkg.json

Co-authored-by: Kai Pastor <dg0yt@darc.de>

saadshams committed:
update

saadshams committed:
update

update to 1.2.0

saadshams committed:
update

update to 1.2.0

LilyWangLL commented on ports/nanojsonc/vcpkg.json:
```diff
@@ -1,10 +1,11 @@
   "description": "Event-Driven JSON Parser for C",
   "homepage": "https://github.com/open-source-patterns/nanojsonc",
   "license": "BSD-3-Clause",
+  "supports": "windows | linux | osx | ios | android | uwp | xbox | emscripten | wasm32",
```
```suggestion
```
Since this port is supported by all aforementioned platforms, please remove this support expression.

LilyWangLL commented on versions/n-/nanojsonc.json:
```diff
@@ -1,9 +1,14 @@
+      "git-tree": "7f713ade1fffaa8477201a7128d2f616fb968047",
       "version": "1.1.0",
-      "port-version": 0
+      "port-version": 1
```
Please apply this suggestion, don't modify here's `"port-version"`.

LilyWangLL commented on versions/baseline.json:
```diff
@@ -6261,8 +6261,8 @@
-      "baseline": "1.1.0",
-      "port-version": 0
+      "baseline": "1.2.0",
+      "port-version": 2
```
```suggestion
      "port-version": 0
```

LilyWangLL commented:
Please try not to manually modify the versions files (`versions/baseline.json` and `versions/n-/nanojsonc.json
`). After you finish updating the port-related codes, run command `.\vcpkg x-add-version nanojsonc --overwrite-version`, which will automatically update the version files. For reference, see the maintainer guide: https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md#versioning.

saadshams committed:
Update versions/baseline.json

Co-authored-by: Lily Wang <94091114+LilyWangLL@users.noreply.github.com>

saadshams committed:
Update versions/n-/nanojsonc.json

Co-authored-by: Kai Pastor <dg0yt@darc.de>

saadshams commented:
remove supports field: OK
Leaving `versions/baseline.json` and `versions/n-/nanojsonc.js` intact. 
ran command `.\vcpkg x-add-version nanojsonc --overwrite-version`
Pushed the code.

dg0yt commented:
Not resolved: Remove the "supports" line.

saadshams committed:
update

update to 1.2.0

saadshams commented:
Sorry. pushed the update without supports field.

LilyWangLL committed:
Update nanojsonc.json

saadshams reviewed (commented):
Thanks. all looks good, the approve field is disabled on my side
Is there an action item for me?

LilyWangLL commented:
> Thanks. for information, how and from where the hash for git-tree is calculated?

You can refer to the documentation https://github.com/MicrosoftDocs/vcpkg-docs/blob/main/vcpkg/users/examples/versioning.getting-started.md#builtin-baseline.

saadshams commented:
Hi @LilyWangLL I mean the following git-tree values, how and from where are they generated from
```

{
  "versions": [
    {
      "git-tree": "41dc847f236000b8a703cf452d37ae36c39ec155",
      "version": "1.2.0",
      "port-version": 0
    },
    {
      "git-tree": "102c65ad71dc556dd82d4af6d83efd96ee316311",
      "version": "1.1.0",
      "port-version": 0
    },
    {
      "git-tree": "7680709fd527db1cc26d47f8897c0669cacbfb0f",
      "version": "1.0.0",
      "port-version": 0
    }
  ]
}

```

dg0yt commented:
This is how it works:
https://learn.microsoft.com/en-us/vcpkg/maintainers/registries#obtain-git-tree-sha
